### PR TITLE
Validation updates

### PR DESCRIPTION
**Pull request by jamesrichard91**

### CODE REVIEW

File: alert-common/src/main/java/com/synopsys/integration/alert/common/security/EncryptionUtility.java [hunk 1]
```diff
@@ -113,7 +113,11 @@ private String getEncodedSalt() {
     }
 
     public boolean isPasswordSet() {
-        return null != getPassword();
+        return StringUtils.isNotBlank(getPassword());
```
Comment (gkillough) [inline alert-common/src/main/java/com/synopsys/integration/alert/common/security/EncryptionUtility.java]: This is a good change, but we should double check that we are clear about blank values not being allowed for these settings in the doc.

File: alert-common/src/main/java/com/synopsys/integration/alert/common/system/BaseSystemValidator.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+import com.synopsys.integration.alert.common.persistence.accessor.SystemMessageUtility;
+
+public class BaseSystemValidator {
+    private SystemMessageUtility systemMessageUtility;
+
+    public BaseSystemValidator(SystemMessageUtility systemMessageUtility) {
+        this.systemMessageUtility = systemMessageUtility;
+    }
+
+    public boolean addSystemMessageIfHasError(String errorMessage, SystemMessageSeverity systemMessageSeverity, SystemMessageType messageType, boolean hasError) {
```
Comment (gkillough) [inline alert-common/src/main/java/com/synopsys/integration/alert/common/system/BaseSystemValidator.java]: I know this method name was my suggestion, but `ForError` might be less clunky than `IfHasError`.

File: src/main/java/com/synopsys/integration/alert/component/settings/SettingsValidator.java [hunk 3]
```diff
@@ -35,45 +31,43 @@
...
+                encryptionUtility.isPasswordMissing());
+            if (encryptionError) {
+                valid = false;
+                logger.error(SettingsDescriptor.FIELD_ERROR_ENCRYPTION_PWD);
+            }
+            boolean saltError = addSystemMessageIfHasError(SettingsDescriptor.FIELD_ERROR_ENCRYPTION_GLOBAL_SALT, SystemMessageSeverity.ERROR, SystemMessageType.ENCRYPTION_CONFIGURATION_ERROR,
+                encryptionUtility.isGlobalSaltMissing());
+            if (saltError) {
+                valid = false;
+                logger.error(SettingsDescriptor.FIELD_ERROR_ENCRYPTION_GLOBAL_SALT);
```
Comment (gkillough) [inline src/main/java/com/synopsys/integration/alert/component/settings/SettingsValidator.java]: We do this a few times, should we add an overloaded version of the method with a `logError` flag?

Comment (jamesrichard91) [reply src/main/java/com/synopsys/integration/alert/component/settings/SettingsValidator.java]: The problem is we are not always logging at error level, some of them are warnings which makes it more complicated.

**Pull request merged**